ckanext/datagovuk/upload.py

```diff
@@ -0,0 +1,150 @@
+'''
+upload.py
+
+Contains functions that upload the organograms to S3.
+
+'''
+import cgi
+import os
+import StringIO
+import zipfile
+import mimetypes
+import collections
+import logging
+import datetime
+from dateutil import parser
+
+from slugify import slugify
+from pylons import config
+import boto3
+import requests
+from botocore.exceptions import ClientError
+
+import paste.fileapp
+import ckan.plugins.toolkit as toolkit
+import ckan.lib.uploader as uploader
+from ckan.common import request
+
+
+def setup_s3_bucket():
+    '''
+    setup_s3_bucket - Grabs the required info from config file and initializes S3 connection
+    '''
+    aws_access_key_id = config.get('ckan.datagovuk.s3_aws_access_key_id')
+    aws_secret_access_key = config.get('ckan.datagovuk.s3_aws_secret_access_key')
+    aws_region_name = config.get('ckan.datagovuk.s3_aws_region_name')
+    if aws_region_name:
+        s3 = boto3.resource('s3',
+                            aws_access_key_id=aws_access_key_id,
+                            aws_secret_access_key=aws_secret_access_key,
+                            region_name=aws_region_name)
+    else:
+        s3 = boto3.resource('s3',
+                            aws_access_key_id=aws_access_key_id,
+                            aws_secret_access_key=aws_secret_access_key)
```

Comment:
```python
kwargs = {
  "aws_access_key_id": aws_access_key_id,
  "aws_secret_access_key": aws_secret_access_key,
}

if aws_region_name:
    kwargs["region_name"] = aws_region_name

s3 = boto3.resource('s3', **kwargs)
```

I think this reads a bit better and avoids potential mistakes with the duplication of the access keys.